command/agent/txn_endpoint.go

```diff
@@ -0,0 +1,227 @@
+package agent
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/consul/structs"
+)
+
+const (
+	// maxTxnOps is used to set an upper limit on the number of operations
+	// inside a transaction. If there are more operations than this, then the
+	// client is likely abusing transactions.
+	maxTxnOps = 500
```

Comment:
Yep - @dadgar had that suggestion so I enforce the existing max size limit over all the values in a txn - you can do 1 max-size key, 2 max/2-size, and so on.